biogeophys/FatesLeafBiophysParamsMod.F90

```diff
@@ -0,0 +1,301 @@
+module FatesLeafBiophysParamsMod
+
+  use FatesConstantsMod , only: r8 => fates_r8
+  use FatesConstantsMod , only: fates_check_param_set
+  use FatesParametersInterface, only : param_string_length
+  use FatesGlobals,   only : fates_log
+  use FatesGlobals,   only : endrun => fates_endrun
+  use shr_log_mod      , only : errMsg => shr_log_errMsg
+  use LeafBiophysicsMod, only : lb_params,btran_on_gs_gs1,btran_on_ag_none
+  use FatesParametersInterface, only : fates_parameters_type
+  ! Register the parameters we want the host to provide, and
+  ! indicate whether they are fates parameters or host parameters
+  ! that need to be synced with host values.
+  use FatesParametersInterface, only : fates_parameters_type, param_string_length
+  use FatesParametersInterface, only : dimension_name_pft, dimension_shape_1d
+  use FatesParametersInterface, only : dimension_shape_scalar, dimension_name_scalar
+  use FatesUtilsMod, only : ArrayNint
+  
+  implicit none
+  private ! Modules are private by default
+  save
+
+  public :: LeafBiophysRegisterParams
+  public :: LeafBiophysReceiveParams
+  public :: LeafBiophysReportParams
+  
+  character(len=*), parameter :: sourcefile = &
+        __FILE__
+
+  integer, parameter  :: lower_bound_pft = 1
+
+contains
+
+  ! =====================================================================================
+
+  subroutine LeafBiophysRegisterParams(fates_params)
+
+
+    class(fates_parameters_type), intent(inout) :: fates_params
+
+    character(len=param_string_length), parameter :: dim_names(1) = (/dimension_name_pft/)
+    integer, parameter :: dim_lower_bound(1) = (/ lower_bound_pft /)
+    character(len=param_string_length) :: name
+    character(len=param_string_length), parameter :: dim_names_scalar(1) = (/dimension_name_scalar/)
+
+
+    ! Register scalars
+
+    name = 'fates_daylength_factor_switch'
+    call fates_params%RegisterParameter(name=name, dimension_shape=dimension_shape_scalar, &
+         dimension_names=dim_names_scalar)
+
+    name = 'fates_leaf_stomatal_model'
+    call fates_params%RegisterParameter(name=name, dimension_shape=dimension_shape_scalar, &
+         dimension_names=dim_names_scalar)
+
+    name = 'fates_leaf_stomatal_assim_model'
+    call fates_params%RegisterParameter(name=name, dimension_shape=dimension_shape_scalar, &
+         dimension_names=dim_names_scalar)
+
+    name = 'fates_leaf_photo_tempsens_model'
+    call fates_params%RegisterParameter(name=name, dimension_shape=dimension_shape_scalar, &
+         dimension_names=dim_names_scalar)
```

Comment:
I really like having the parameter registration broken out into "domain" specific modules.  I'm just flagging here that there are being migrated to the host land model in #1300.  Given this and that there are new parameters, I'm now thinking we should bring this in prior to #1300 as it might be easier to deconflict in that order and would allow us to get the new parameters into the update associated with ESCOMP/ctsm#2904.

Reply:
Just updating here for posterity that we talked about this and we're currently planning to bring this in after #1300.